File: 0027.移除元素/solution.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int sz=nums.size();
        int s=0,f=0;
        while (f<sz) {
            if (nums.at(f)==val) {
                f++;
            } else {
                nums.at(s)=nums.at(f);
                s++;
                f++;
            }
        }
        return s;
    }
};

File: 0027.移除元素/solution.go
func removeElement(nums []int, val int) int {
	sz := len(nums)
	s, f := 0, 0
	for f < sz {
		if nums[f] == val {
			f++
		} else {
			nums[s] = nums[f]
			s++
			f++
		}
	}
	return s
}